toolbox

```diff
@@ -477,6 +478,105 @@ list_containers()
 )
 
 
+run_rm()
+(
+    force_remove=$1
+    command_to_run=$2
+    id_or_name=$3
+    keyword=$4
+
+    if $force_remove; then
+        if ! rm_output=$($prefix_sudo podman $command_to_run $id_or_name --force 2>&42); then
+            echo "$0: failed to remove $id_or_name"
+            exit 1
+        fi
+    else
+        if ! rm_output=$($prefix_sudo podman $command_to_run $id_or_name 2>&42); then
+            echo "$0: failed to remove $id_or_name"
+            exit 1
+        fi
+    fi
+    if ! [ "$rm_output" = "" ]; then
+        echo -e "${LRC}$keyword $rm_output removed${NC}"
+    fi
+)
+
+
+remove_image()
+(
+    id_or_name_list=$1
+    force_remove=$2
+    remove_all=$3
+
+    if ! $remove_all; then
+        IFS=' '
+        export IFS
+        for id_or_name in $id_or_name_list; do
+            if ! inspect_out=$($prefix_sudo podman inspect $id_or_name --type image --format \"{{.Labels}}\" 2>&42); then
+                echo "$0: failed to inspect $id_or_name"
+                exit 1
+            fi
+            if ! [[ $inspect_out = *"com.redhat.component:fedora-toolbox"* ]]; then
+                echo "Not a fedora-toolbox image"
+                exit 1
```

Comment:
We should probably match the internal behaviour of `podman rm` here. See `man podman-rm` or `man podman-rmi` [1].

In short, when multiple IDs or `--all` are passed, it goes through all of them and tries to remove as many as it can, even if some of them couldn't be removed. However, the exit code is only 0 if all the IDs were successfully removed. To do that, we need to use `return`, not `exit` here.

[1] There's a bug in Podman: https://github.com/containers/libpod/issues/2539

Reply:
Done.